src/util-inl.h

```diff
@@ -229,30 +230,85 @@ bool StringEqualNoCaseN(const char* a, const char* b, size_t length) {
   return true;
 }
 
+// clang
+#ifndef __has_builtin
+#define __has_builtin(x) (0)
+#endif
+
+template<typename T>
+T MultiplyWithOverflowCheck(T a, T b) {
+  T ret;
+
+#if NODE_GNUC_AT_LEAST(5, 0, 0) || __has_builtin(__builtin_mul_overflow)
+  CHECK(!__builtin_mul_overflow(a, b, &ret));
+#else
+  ret = a * b;
+  if (a != 0)
+    CHECK_EQ(b, ret / a);
```

Comment:
> Wouldn't `CHECK_GE(ret, a)` be sufficient?

No, unfortunately not. `uint8` example: in `140 * 3 == 164` the multiplication wraps around, but a `CHECK_GE(164, 140)` would pass.